lib/calendar.ts

```diff
@@ -1185,15 +1111,35 @@ const helperSharedImpl: HelperSharedImpl = {
       throw new RangeError(`Day ${day} does not exist in resulting calendar month`);
     }
     return calendarDate;
-  },
-  addCalendar(calendarDate, { years = 0, months = 0, weeks = 0, days = 0 }, overflow, cache) {
+  }
+  addCalendar(
+    calendarDate: CalendarYMD,
+    {
+      years,
+      months,
+      weeks,
+      days
+    }: {
+      years?: number | undefined;
+      months?: number | undefined;
+      weeks?: number | undefined;
+      days?: number | undefined;
+    },
+    overflow: Overflow,
+    cache: OneObjectCache
+  ): FullCalendarDate {
     const { year, month, day } = calendarDate;
-    const addedMonths = this.addMonthsCalendar({ year: year + years, month, day }, months, overflow, cache);
-    const initialDays = days + weeks * 7;
+    const addedMonths = this.addMonthsCalendar({ year: year + (years ?? 0), month, day }, months ?? 0, overflow, cache);
```

Comment:
I believe this (and the same change in the line below) was the only runtime change found in this PR. I'll plan to port it over to proposal-temporal.

Reply:
It seemed to me like the previous code was actually better! (Besides the fact of not being typed, that is...)
I guess if we were to do `{ years = 0 }: { years?: number | undefined }` TypeScript is not smart enough to figure that the `years` that you get in the body of the function can only be a `number`?

By the way, do we need `years?: number | undefined` rather than just `years?: number`? It seems like the question mark operator should cover that.

Reply:
You're exactly right. I think what happened here was that an intermediate state of this class had an externally-declared `interface` and I couldn't figure out how to get the `= 0` defaults to work with that external declaration. But later I removed that declaration it for DRY reasons (now just relying on the class itself) but forgot to revert the default params changes.  Latest commit restores the old code which I agree was better. Thanks for catching this!